yarn-project/aztec.js/src/contract/deploy_method.ts

```diff
@@ -248,15 +265,26 @@ export class DeployMethod<TContract extends ContractBase = Contract> extends Bas
   }
 
   /**
-   * Estimates gas for the interaction and returns gas limits for it.
-   * @param options - Options.
-   * @returns Gas limits.
+   * Simulate the deployment
+   *
+   * @param options - An optional object containing additional configuration for the simulation.
+   * @returns A simulation result object containing metadata of the execution, including gas
+   * estimations (if requested via options), execution statistics and emitted offchain effects
    */
-  public override async estimateGas(
-    options: Omit<DeployOptions, 'estimateGas'>,
-  ): Promise<Pick<GasSettings, 'gasLimits' | 'teardownGasLimits'>> {
+  public async simulate(options: SimulateDeployOptions): Promise<SimulationReturn<true>> {
     const executionPayload = await this.request(options);
-    return this.wallet.estimateGas(executionPayload, options);
+    const simulatedTx = await this.wallet.simulateTx(executionPayload, options);
+
+    const { gasLimits, teardownGasLimits } = getGasLimits(simulatedTx, options.fee?.estimatedGasPadding);
+    this.log.verbose(
+      `Estimated gas limits for tx: DA=${gasLimits.daGas} L2=${gasLimits.l2Gas} teardownDA=${teardownGasLimits.daGas} teardownL2=${teardownGasLimits.l2Gas}`,
+    );
+    return {
+      stats: simulatedTx.stats!,
+      offchainEffects: collectOffchainEffects(simulatedTx.privateExecutionResult),
+      result: undefined,
```

Comment:
```suggestion
      result: undefined, // Constructors never return any value
```

Would this ^ be correct? The `result` here is a return value, right?